lib/matplotlib/image.py

```diff
@@ -1582,6 +1584,10 @@ def imsave(fname, arr, vmin=None, vmax=None, cmap=None, format=None,
             _api.check_in_list(('upper', 'lower'), origin=origin)
         if origin == "lower":
             arr = arr[::-1]
+
+        if (isinstance(arr, list)):
+            arr = np.asarray(arr, dtype=np.uint8)
+
```

Comment:
Let's move the format coersion above the origin handling. While the reversal (`[::-1]`) also works for lists, it's more efficient and easier to understand if we ensure that we have the array already there.

Also `arr = np.asanyarray(array)` should be enough.
- `asanyarray()` instead of `asarray()` Passing through array subclasses should be ok - we currently do that as well.
- let's not do dtype conversion in this PR - we currently also don't do this for arrays. The user is responsible for passing reasonable types. Any change to that would be a separate topic.